### ISSUE TITLE
oauth not working SSL Errors

### ISSUE THREAD

**Issue by ashwarsadh** [burnash/gspread#994]
Since few days oAuth has stopped working and I am just getting SSL Errors

```
Traceback (most recent call last):
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
    conn.connect()
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/connection.py", line 416, in connect
    self.sock = ssl_wrap_socket(
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib/python3.8/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.8/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:1131)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ashwar/.local/lib/python3.8/site-packages/requests/adapters.py", line 440, in send
    resp = conn.urlopen(
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/home/ashwar/.local/lib/python3.8/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='oauth2.googleapis.com', port=443): Max retries exceeded with url: /token (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "po.py", line 4, in <module>
    gc = gspread.oauth(credentials_filename='./client_secret.json')
  File "/home/ashwar/.local/lib/python3.8/site-packages/gspread/auth.py", line 161, in oauth
    creds = flow(client_config=client_config, scopes=scopes)
  File "/home/ashwar/.local/lib/python3.8/site-packages/gspread/auth.py", line 65, in local_server_flow
    return flow.run_local_server(port=port)
  File "/home/ashwar/.local/lib/python3.8/site-packages/google_auth_oauthlib/flow.py", line 477, in run_local_server
    self.fetch_token(authorization_response=authorization_response)
  File "/home/ashwar/.local/lib/python3.8/site-packages/google_auth_oauthlib/flow.py", line 286, in fetch_token
    return self.oauth2session.fetch_token(self.client_config["token_uri"], **kwargs)
  File "/home/ashwar/.local/lib/python3.8/site-packages/requests_oauthlib/oauth2_session.py", line 341, in fetch_token
    r = self.request(
  File "/home/ashwar/.local/lib/python3.8/site-packages/requests_oauthlib/oauth2_session.py", line 521, in request
    return super(OAuth2Session, self).request(
  File "/home/ashwar/.local/lib/python3.8/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/ashwar/.local/lib/python3.8/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/home/ashwar/.local/lib/python3.8/site-packages/requests/adapters.py", line 517, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='oauth2.googleapis.com', port=443): Max retries exceeded with url: /token (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:1131)')))
```

I have tried following
Update OpenSSL
update python to 3.10
update all relevant Python modules including requests, urllib3, oauth2client, google-api-python-client
Upgrade Linux Distro
Installed Fresh Ubuntu Distro on a new system and tried everything
Read up all possible solutions for 2 days over internet and tried all.

Seems like Google has updated some SSL/ TLS Requirements, although my setup supports TLS1.3 but still google is rejecting Login.
Please advise solution

**Comment by lavigne958** [comment]
Hello @ashwarsadh 

thank you for raising this issue, I will look at it closely as soon as I can.

**Comment by lavigne958** [comment]
Hello I am sorry to back to you with bad news.

I can't reproduce your issue.
I tried it on 2 different computers, with very different Linux distributions.

I tried different version of `google-auth-oauthlib` everything works everytimes.

What I can advise you to do is:
1. remove the following libraries, because they are not necessary for gspread to work:
  a. oauth2client
  b. google-api-python-client
2. make sure you have the necessary dependencies installed in their latest versions:
  a. google-auth-oauthlib
  b. google-auth

Hope this helps.

Go on stackoverflow using the gspread tag to get help from others, they will surely be able to help you solve your problem according to your exact setup.

In mean time I will close this issue, feel free to comment if find that this is an actual bug in gspread source code.
